pvlib/modelchain.py

```diff
@@ -622,14 +622,14 @@ def infer_spectral_model(self):
             return self.sapm_spectral_loss
         elif ((('Technology' in params or
                 'Material' in params) and
-               (pvsystem._infer_cell_type() is not None)) or
+               (self.system._infer_cell_type() is not None)) or
               'first_solar_spectral_coefficients' in params):
             return self.first_solar_spectral_loss
         else:
             raise ValueError('could not infer spectral model from '
                              'system.module_parameters. Check that the '
                              'parameters contain valid '
-                             'first_solar_spectral_coefficients or a valid '
+                             'first_solar_spectral_coefficients and a valid '
```

Comment:
I still think this should be `or`.  If `Technology` or `Material` is not in `params`, then attempt to identify the cell type.  _Or_ find 'first_solar_spectral_coefficients`.

Reply:
Thanks @cwhanse for catching this. I agree that `or` describes the logic. @jgaffiot can you revert this change?

Unrelated to this change, but the error message also neglects the possibility of using `sapm_spectral_loss`.

Reply:
I changed the comment because the code says 'and' line 624:

         elif ((('Technology' in params or
                'Material' in params) and
               (pvsystem._infer_cell_type() is not None)) or
               (self.system._infer_cell_type() is not None)) or
              'first_solar_spectral_coefficients' in params):

Reply:
I change the code too ?